internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1928,6 +1958,12 @@ func resourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{})
 			if err := d.Set("api_server_authorized_ip_ranges", apiServerAuthorizedIPRanges); err != nil {
 				return fmt.Errorf("setting `api_server_authorized_ip_ranges`: %+v", err)
 			}
+			if accessProfile.EnableVnetIntegration != nil {
+				enableAPIServerVnetIntegration = *accessProfile.EnableVnetIntegration
+			}
+			if accessProfile.SubnetId != nil && *accessProfile.SubnetId != "" {
+				d.Set("api_server_subnet_id", accessProfile.SubnetId)
+			}
```

Comment:
This should always have a value set into state so can this become 
```suggestion
			apiServerSubnetId := ""
			if accessProfile.SubnetId != nil {
				apiServerSubnetId = *accessProfile.SubnetId
			}
			d.Set("api_server_subnet_id", apiServerSubnetId)
```